core/src/main/java/io/opencensus/stats/AggregationData.java

```diff
@@ -88,6 +76,68 @@ static SumData create(double sum) {
         Function<? super AggregationData, T> defaultFunction) {
       return p0.apply(this);
     }
+
+    /** Applies the given match function to the underlying data type. */
+    public abstract <T> T match(
+        Function<? super SumDataDouble, T> p0,
+        Function<? super SumDataLong, T> p1,
+        Function<? super SumData, T> defaultFunction);
+
+    /** The double sum value on aggregated values of type {@code MeasureDouble}. */
+    @Immutable
+    @AutoValue
+    public abstract static class SumDataDouble extends SumData {
```

Comment:
As discussed irl, I think it makes sense just to inline these types all in AggregationData - it doesn't increase the overall number of things we need to deal with and removes the need to deal with additional defaultFunctions.